### PR TITLE
Auditing Composition UI

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Refactor Auditing to Vue3 composition.

# Delete branch after merge
YES 

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## Enhancements
* Advanced search and CSV export in auditing

### CODE REVIEW

File: html/pfappserver/root/src/views/Auditing/_router/index.js [hunk 1]
```diff
@@ -1,179 +1,27 @@
...
-import RadiusLogsStore from '../_store/radiusLogs'
-import DhcpOption82LogsStore from '../_store/dhcpOption82Logs'
-import DnsLogsStore from '../_store/dnsLogs'
-import AdminApiAuditLogs from '../_store/adminApiAuditLogs'
-import LiveLogs from '../_store/liveLogs'
+import AdminApiLogsRoutes from '../adminApiLogs/_router'
+import DhcpOption82LogsRoutes from '../dhcpOption82Logs/_router'
+import DnsLogsRoutes from '../dnsLogs/_router'
+import RadiusLogsRoutes from '../radiusLogs/_router'
+import LiveLogsRoutes from '../liveLogs/_router'
```
Comment (cgx) [inline html/pfappserver/root/src/views/Auditing/_router/index.js:6]: This file is missing.

Comment (satkunas) [reply html/pfappserver/root/src/views/Auditing/_router/index.js:6]: Missing file added in a4f46426f4a23dbca9ff28c0e109ee1bc541c435

File: html/pfappserver/root/src/views/Auditing/radiusLogs/_components/TheSearch.vue [hunk 2]
```diff
@@ -1,3 +1,153 @@
...
+// @vue/component
+export default {
+  name: 'the-search',
+  inheritAttrs: false,
+  components,
+  setup
+}
+</script>
+
+<!--
```
Comment (cgx) [inline html/pfappserver/root/src/views/Auditing/radiusLogs/_components/TheSearch.vue]: Can you remove the commented-out code?

File: html/pfappserver/root/src/views/Auditing/radiusLogs/_components/TheView.vue [hunk 3]
```diff
@@ -0,0 +1,140 @@
+<template>
+  <b-form @submit.prevent ref="rootRef">
+    <b-card no-body>
+      <b-card-header>
+        <b-button-close @click="close" v-b-tooltip.hover.left.d300 :title="$t('Close [ESC]')"><icon name="times"></icon></b-button-close>
```
Comment (cgx) [inline html/pfappserver/root/src/views/Auditing/radiusLogs/_components/TheView.vue]: `close` doesn't seem to be defined.

**Comment by cgx** [comment]
Listing the search criteria is useless when listing *all* possible fields.

placeholders-storm

**Comment by cgx** [comment]
The initial state of the advanced search is no longer matching the criteria of the simple search — I understand that it could be intense since we perform a search on *all* possible fields. However, the padding of the form input fields seems wrong.

advanced-search-initial-state

**Comment by cgx** [comment]
When I look at a single DNS log entry, I get an *undefined* error. Could be related to my dataset.

dns-audit-log-undefined-identifier

**Comment by cgx** [comment]
The toggle button on the live logs page is misaligned with the label.

live-logs-re-alignment

**Comment by cgx** [comment]
The save searches that appear under the search button and under the navigation item don't match.

save-search-2
save-search-1

**Comment by satkunas** [comment]
> Listing the search criteria is useless when listing _all_ possible fields.
> 
> placeholders-storm

I've reworked the placeholder to show a maximum of 3 item, while also adding a title (w/ hover) that expands all the criteria. 

Screen Shot 2021-07-06 at 10 38 06 AM

**Comment by satkunas** [comment]
> The toggle button on the live logs page is misaligned with the label.
> 
> live-logs-re-alignment

Fixed in 0b008f8c4792ff94d87c24261f8c5e9c0a2819ea

**Comment by satkunas** [comment]
> The initial state of the advanced search is no longer matching the criteria of the simple search — I understand that it could be intense since we perform a search on _all_ possible fields. However, the padding of the form input fields seems wrong.
> 
> advanced-search-initial-state

I don't like expanding all the available criteria as the default search condition. Instead I added a placeholder on a single field using the ID (or MAC).

**Comment by satkunas** [comment]
> When I look at a single DNS log entry, I get an _undefined_ error. Could be related to my dataset.
> 
> dns-audit-log-undefined-identifier

I'm not able to reproduce. This appears to be your dataset.

**Comment by satkunas** [comment]
> The save searches that appear under the search button and under the navigation item don't match.
> 
> save-search-2
> save-search-1



This will be resolved later. I still need to unify the 3 user-preference store modules into a single module (either VueX or Pinia), as well as Vue3-ize all the index.vue for the sidebar content. I will resolve both at the same time. I will also try to find a more elegant way to display the saved searches in the sidebar without using as much vertical space. One idea is to include a save-search icon in the sidebar item that doubles as a dropdown containing all the save searches. Thoughts?

**Comment by satkunas** [comment]
@cgx all comments addressed. Please follow-up.

**Comment by cgx** [comment]
> I will also try to find a more elegant way to display the saved searches in the sidebar without using as much vertical space. One idea is to include a save-search icon in the sidebar item that doubles as a dropdown containing all the save searches. Thoughts?

I think it is convenient to have the saved searches accessible from the sidebar. Having them visible without having to expand a list also saves a mouse click. To save space, we could remove the *Saved Searches* header and put a small magnifier icon on the left side of each saved search.

**Comment by cgx** [comment]
> I've reworked the placeholder to show a maximum of 3 item, while also adding a title (w/ hover) that expands all the criteria.
> 
> Screen Shot 2021-07-06 at 10 38 06 AM

I wonder if this is just too much information? Since the search will cover all fields, the results should be no surprise for the user. The placeholder could simply be *Search criteria*.

**Comment by cgx** [comment]
> I don't like expanding all the available criteria as the default search condition. Instead I added a placeholder on a single field using the ID (or MAC).

The padding is still incorrect (missing) between the input fields.

**Comment by satkunas** [comment]
> > I've reworked the placeholder to show a maximum of 3 item, while also adding a title (w/ hover) that expands all the criteria.
> > Screen Shot 2021-07-06 at 10 38 06 AM
> 
> I wonder if this is just too much information? Since the search will cover all fields, the results should be no surprise for the user. The placeholder could simply be _Search criteria_.

I've reworked the placeholder and tooltip as a compromise.

Screen Shot 2021-07-07 at 1 30 58 PM
Screen Shot 2021-07-07 at 1 31 26 PM

**Pull request merged**